contrib/devtools/github-merge.py

```diff
@@ -105,9 +105,23 @@ def main():
     host_repo = host+":"+repo # shortcut for push/pull target
 
     # Extract settings from command line
-    args = parse_arguments(branch)
+    args = parse_arguments()
     pull = str(args.pull[0])
+
+    # Receive pull information from github
+    info = retrieve_pr_info(repo,pull)
+    if info is None:
+        exit(1)
+    title = info['title']
+    # precedence order for destination branch argument:
+    #   - command line argument
+    #   - githubmerge.branch setting
+    #   - base branch for pull (as retrieved from github)
+    #   - 'master'
     branch = args.branch
+    branch = opt_branch if branch is None else branch
+    branch = info['base']['ref'] if branch is None else branch
+    branch = 'master' if branch is None else branch
```

Comment:
That's much shorter, also seems to work:
```
>>> None or 'a' or 'b' or 'c'
'a'
>>> None or None or 'b' or 'c'
'b'
>>> None or None or None or 'c'
'c'
>>> 'f' or None or None or 'c'
'f'
```
I think the subtle difference in this case is that '' is seen as False (hence the use of `is None`. But doesn't matter here, an empty branch name is pointless.